src/Features/Core/Completion/AbstractCompletionService.CompletionRules.cs

```diff
@@ -41,7 +41,7 @@ protected PatternMatcher GetPatternMatcher(string value)
                 // MRU list, then we definitely want to include it.
                 if (filterText.Length == 0)
                 {
-                    if (item.Preselect || _completionService.GetMRUIndex(item) < 0)
+                    if (item.Preselect || _completionService.ShouldPreselect(item))
```

Comment:
The completion service no longer exposes the idea that it has any sort of MRU.  Instead, we just ask it relevant questions, and it can answer them however it wants to answer them.